BrainPortal/app/models/data_provider.rb

```diff
@@ -1353,7 +1353,43 @@ def self.cleanup_leftover_cache_files(do_it=false, options={})
     end
   end
 
+  # Updates the time stamp for important auxiliary directories and files
+  # as workaround for HPC file deletion policies.
+  #
+  # Some Bourreaux systems are configured with disk allocations where files older than N days are erased automatically.
+  # To prevent such system from deleting the top-level directories for the DP_Cache, and some cbrain-specific files,
+  # the boot process should touch them to reset their timestamps.
+  #
+  # For a portal or bourreau:
+  #
+  # - the +DataProvider+ cache dir
+  # - the +DP_Cache_Key.md5+ and
+  # - +DP_Cache_Rev.id+ located in that cache dir
+  #
+  # For a bourreau:
+  #
+  # - the +gridshare+ dir
+  # - the +DP_Cache+ symbolic link located in it.
+  def self.system_touch
+    myself       = RemoteResource.current_resource
+    cache_dir    = myself.dp_cache_dir
+    dp_cache_id  = File.join cache_dir, DataProvider::DP_CACHE_ID_FILE
+    dp_cache_md5 = File.join cache_dir, DataProvider::DP_CACHE_MD5_FILE
+
+    FileUtils.touch [cache_dir, dp_cache_id, dp_cache_md5], verbose: true, nocreate: true
 
+    # touch only cache for Portal, for Bourreau touch gridshare
+    return true unless myself.is_a? Bourreau
+
+    gridshare_dir = myself.cms_shared_dir
```

Comment:
Please move the logic of lines 1384-1391 to the file `Bourreau/lib/bourreau_system_checks.rb` within the method `self.a100_ensure_dp_cache_symlink_exists` like I asked in my previous comments.